### ISSUE TITLE
Alert configuration state is not stable

### ISSUE THREAD

**Issue by gdelbergue** [mongodb/terraform-provider-mongodbatlas#632]
Hello,

We use `mongodbatlas_alert_configuration` to setup our alerts with Opsgenie. It works fine. However, every time we plan / apply without any changes of HCL, terraform detects changes. Moreover, even if we apply, issue remains. I guess resource is not well set or state storing has an issue.

### Terraform CLI and Terraform MongoDB Atlas Provider Version

```hcl
Terraform: 1.0.6
Provider Atlas: 1.1.1
```

### Terraform Configuration File 

```hcl
resource "mongodbatlas_alert_configuration" "host_alerts" {
  for_each   = toset(["HOST_DOWN", "NO_PRIMARY", "HOST_MONGOT_CRASHING_OOM"])
  project_id = var.project_id
  event_type = each.key
  enabled    = true

  notification {
    ops_genie_api_key = var.api_token
    ops_genie_region  = "EU"
    type_name         = "OPS_GENIE"
    interval_min      = 5
    delay_min         = 0
    sms_enabled       = false
    email_enabled     = true
    roles             = ["GROUP_CHARTS_ADMIN", "GROUP_CLUSTER_MANAGER"]
  }
}
```

As we can see, `email_enabled` is set as well as `interval_min` and `roles`.

### Steps to Reproduce

1. `terraform init`
2. `terraform apply`

### Expected Behavior

`interval_min` is set as well as `email_enabled` and `roles` and a plan/apply doesn't trigger changes after change has been set.

### Actual Behavior

```
resource "mongodbatlas_alert_configuration" "host_alerts" {
        id                     = "xxxxxx"
        # (6 unchanged attributes hidden)

      ~ notification {
          ~ email_enabled          = false -> true
          ~ interval_min           = 0 -> 5
          ~ roles                  = [
              + "GROUP_CHARTS_ADMIN",
              + "GROUP_CLUSTER_MANAGER",
            ]
            # (19 unchanged attributes hidden)
        }
    }
```

**Comment by themantissa** [comment]
@guyguy333 let me have the team take a look.

**Comment by themantissa** [comment]
Internal ticket INTMDB-284

**Comment by themantissa** [comment]
@guyguy333 we determined the root cause.  You cannot include both sms_enabled and email_enabled unless type_name is GROUP, ORG, or USER.  The API will accept the parameters without error but it's why the state mismatches later.  However the documentation is unclear on this so we will update that.

**Comment by gdelbergue** [comment]
Thanks @themantissa. I read the updated documentation in PR. So yes, in that case, as we're using Opsgenie, we can remove both fields `sms_enabled` and `email_enabled`. It now ends with:

```
~ notification {
          ~ interval_min           = 0 -> 5
          ~ roles                  = [
              + "GROUP_CHARTS_ADMIN",
              + "GROUP_CLUSTER_MANAGER",
            ]
            # (20 unchanged attributes hidden)
        }

```

I think I understand roles issue according to updated documentation and if I understand right, it has no impact for Opsgenie and so should be removed. However, for interval_min, I don't get why this diff is remaining:

```
notification {
          ~ interval_min           = 0 -> 5
            # (21 unchanged attributes hidden)
        }

```


 ```hcl
 resource "mongodbatlas_alert_configuration" "host_alerts" {
  for_each   = toset(["HOST_DOWN", "NO_PRIMARY", "HOST_MONGOT_CRASHING_OOM"])
  project_id = var.project_id
  event_type = each.key
  enabled    = true

  notification {
    ops_genie_api_key = var.api_token
    ops_genie_region  = "EU"
    type_name         = "OPS_GENIE"
    interval_min      = 5
    delay_min         = 0
  }
}
```

Thanks

**Comment by themantissa** [comment]
@guyguy333 that's actually a slightly different issue corrected in https://github.com/mongodb/terraform-provider-mongodbatlas/pull/624.  Will be included in the next release.

**Comment by themantissa** [comment]
Fixed in 1.2
